frontend/src/screens/dashboard/docsqa/main/context.tsx

```diff
@@ -0,0 +1,303 @@
+            setAnswer((prevAnswer) => prevAnswer + parsed.content)
+            setIsRunningPrompt(false)
+          } else if (parsed?.type === 'docs') {
+            setSourceDocs((prevDocs) => [...prevDocs, ...parsed.content])
+          }
+        } catch (err: any) {
+          throw new Error('An error occurred while processing the response.')
+        }
+      })
+
+      sseRequest.addEventListener('end', (event: any) => {
+        sseRequest.close()
+      })
+
+      sseRequest.addEventListener('error', (event: any) => {
+        sseRequest.close()
+        setPrompt('')
+        setIsRunningPrompt(false)
+        const message = JSON.parse(event.data).detail[0].msg
+        notifyError('Failed to retrieve answer', { message })
+      })
+    } catch (err: any) {
+      setPrompt('')
+      setIsRunningPrompt(false)
+      notifyError('Failed to retrieve answer', err)
+    }
+  }
+
+  const createChatApplication = async (
+    applicationName: string,
+    questions: [],
+    setApplicationName: (name: string) => void,
+  ) => {
+    if (!applicationName) {
+      return notify('error', 'Application name is required')
+    }
+    const selectedModel = allEnabledModels.find(
+      (model: any) => model.name == selectedQueryModel,
+    )
+
+    try {
+      await createApplication({
+        name: `${applicationName}-rag-app`,
+        config: {
+          collection_name: selectedCollection,
+          model_configuration: {
+            name: selectedModel.name,
+            provider: selectedModel.provider,
+            ...JSON.parse(modelConfig),
+          },
+          retriever_name: selectedRetriever?.name ?? '',
+          retriever_config: JSON.parse(retrieverConfig),
+          prompt_template: promptTemplate,
+          query_controller: selectedQueryController,
+        },
+        questions,
+      }).unwrap()
+      setApplicationName('')
+      setIsCreateApplicationModalOpen(false)
+      notify('success', 'Application created successfully')
+    } catch (err: any) {
+      notify('error', 'Failed to create application', err?.data?.detail)
+    }
+  }
+
+  const resetQA = () => {
+    setAnswer('')
+    setErrorMessage(false)
+    setPrompt('')
+  }
+
+  useEffect(() => {
+    if (collections && collections.length) setSelectedCollection(collections[0])
+
+    if (allQueryControllers && allQueryControllers.length)
+      setSelectedQueryController(allQueryControllers[0])
+
+    if (allEnabledModels && allEnabledModels.length)
+      setSelectedQueryModel(allEnabledModels[0].name)
+
+    if (allRetrieverOptions && allRetrieverOptions.length) {
+      setSelectedRetriever(allRetrieverOptions[0])
+      setPromptTemplate(allRetrieverOptions[0].promptTemplate)
+    }
+    if (selectedRetriever)
+      setRetrieverConfig(JSON.stringify(selectedRetriever.config, null, 2))
+  }, [
+    collections,
+    allQueryControllers,
+    allEnabledModels,
+    allRetrieverOptions,
+    selectedRetriever,
```

Comment:
Are these interdependent, can we break this use-effect if not?

Reply:
we can use multiple use effect as well they are not connected

Reply:
Did we not end up addressing this?

Reply:
Should we keep different useEffect for each setUpdate function i thought we can do it in one use effect thats why